modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/OnboardingUpgradeFeaturesPage.kt

```diff
@@ -111,69 +138,400 @@ internal fun OnboardingUpgradeFeaturesPage(
+                canUpgrade = canUpgrade,
+            )
+        }
+    }
+}
 
-        OnboardingUpgradeHelper.PlusBackground(Modifier.verticalScroll(scrollState)) {
-            Column(
-                Modifier
-                    .windowInsetsPadding(WindowInsets.statusBars)
-                    .windowInsetsPadding(WindowInsets.navigationBars)
-                    .heightIn(min = this@BoxWithConstraints.calculateMinimumHeightWithInsets())
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun BoxWithConstraintsScope.UpgradeLayout(
+    state: OnboardingUpgradeFeaturesState,
+    scrollState: ScrollState,
+    onBackPressed: () -> Unit,
+    onNotNowPressed: () -> Unit,
+    onSubscriptionFrequencyChanged: (Int) -> Unit,
+    onFeatureCardChanged: (Int) -> Unit,
+    onUpgradePressed: () -> Unit,
+    canUpgrade: Boolean,
+    upgradePrice: (String) -> String,
+) {
+    OnboardingUpgradeHelper.UpgradeBackground(
+        modifier = Modifier.verticalScroll(scrollState),
+        shortNamRes = state.currentFeatureCard.shortNameRes,
+        backgroundGlowsRes = state.currentFeatureCard.backgroundGlowsRes,
+    ) {
+        Column(
+            Modifier
+                .windowInsetsPadding(WindowInsets.statusBars)
+                .windowInsetsPadding(WindowInsets.navigationBars)
+                .heightIn(min = this.calculateMinimumHeightWithInsets()),
+        ) {
+
+            Spacer(Modifier.height(8.dp))
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
             ) {
-
-                Spacer(Modifier.height(8.dp))
                 NavigationIconButton(
                     onNavigationClick = onBackPressed,
                     iconColor = Color.White,
                     modifier = Modifier
                         .height(48.dp)
                         .width(48.dp)
                 )
+                TextH30(
+                    text = stringResource(LR.string.not_now),
+                    color = Color.White,
+                    modifier = Modifier
+                        .padding(horizontal = 24.dp)
+                        .clickable { onNotNowPressed() },
+                )
+            }
 
-                Spacer(Modifier.height(12.dp))
+            Spacer(Modifier.weight(1f))
+
+            Column {
+                AnimatedContent(
+                    targetState = state.currentFeatureCard.titleRes,
+                    label = "titleRes"
+                ) { titleRes ->
+                    TextH20(
+                        text = stringResource(titleRes),
+                        textAlign = TextAlign.Center,
+                        color = Color.White,
+                        modifier = Modifier
+                            .padding(horizontal = 24.dp)
+                            .fillMaxWidth(),
+                    )
+                }
```

Comment:
The default animation for the prices does feel a bit jarring to me. I also wonder if it might be worth trying to give a matching animation to the year/month text by the price to go along with the price animation.

This is just a nice-to-have for me, feel free to leave it as-is for now if you'd like since it can always be addressed later when someone who knows more about design takes a look. 🙂